### ISSUE TITLE
Salmon quant reports transcripts in BAM not found in FASTA, despite GFFREAD correction

### ISSUE THREAD

**Issue by majdabdul** [COMBINE-lab/salmon#863]
Hello, I hope you're well!

**Context:** I performed short-read (Illumina, 150bp PE) and long-read (ONT, cDNA+PCR) RNA-sequencing on 12 human cell line samples (4 conditions, 3 biological replicates each). I aligned the short-reads to chm13v2.0_RefSeq_Liftoff_v5.1.gff3 using HISAT2, and the long reads to chm13v2.0.fa using Minimap2. Next, I used StringTie hybrid assembly mode (--mix option) to generate GTFs using both technologies (to take advantage of structural information from long reads and accuracy from short reads). I then ran stringtie --merge to generate the transcriptome assembly GTF from the sample GTFs. I took this merged transcriptome and further annotated it using SQANTI3 (with chm13v2.0_RefSeq_Liftoff_v5.1.gtf and chm13v2.0.fa). I converted both the SQANTI-annotated and the StringTie GTFs into fasta as follows:
`gffread -w merged_transcriptome.fa -g chm13v2.0.fa merged_transcripts.gtf`

Now I'm trying to quantify with Salmon.

Firstly, I'm not sure whether it's better to use the short or long reads here, given that my goal is to identify short peptides (I have peptidomics data) derived from specific splicing events. You may or may not be able to help me with that, but if you have thoughts, I'd appreciate them! Anyway, I decided arbitrarily to go with the long reads.

**Bug description:** Secondly, I keep running into:
`Transcript NM_032515.5 appears in the reference but did not appear in the BAM`
and
'Transcript chr19 appeared in the BAM header, but was not in the provided FASTA file'

This is the salmon command I ran:
`$salmon quant --ont -t $transcriptome -l SF -a $bam -o $outdir/$name`

The command failed because there are transcripts in the BAM files not in the FASTA (and vice versa).

As suggested, I used gffread to generate a new transcriptome fasta as follows:
`gffread -w salmon_fix.fa -g chm13v2.0.fa chm13v2.0_RefSeq_Liftoff_v5.1.gff3`

I reran the above salmon command using this new fasta file, but got the same error and warnings (there are both transcripts in the BAM not in the fasta and vice versa).

**Software information:** I used salmon v1.10.0 (though I also tried v1.10.1). In the case of v1.10.0, I had downloaded the pre-compiled binary, and in the case of v1.10.1, the admins of the HPC cluster I used installed it- not sure how. Regardless, all the runs were on HPC clusters, which run on Linux CentOS (I use two different HPC clusters, depending on their availability).

Cluster1:
`$ uname -a
Linux rescomp1.hpc.in.bmrc.ox.ac.uk 3.10.0-1160.92.1.el7.x86_64 #1 SMP Tue Jun 20 11:48:01 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -a
-bash: lsb_release: command not found`

Cluster2:
`$ uname -a
Linux htc-login02 4.18.0-348.7.1.el8_5.x86_64 #1 SMP Wed Dec 22 13:25:12 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -a
LSB Version:    :core-4.1-amd64:core-4.1-noarch
Distributor ID: CentOS
Description:    CentOS Linux release 8.5.2111
Release:        8.5.2111
Codename:       n/a`

**Comment by YIGUIz** [comment]
Hi, I encountered the same issue. Have you managed to solve it?

**Comment by majdabdul** [comment]
Yes, but it was a while ago. I think I had realised that I aligned to the
genome (Step 2 in my original post), when Salmon documentation specifically
says you should align to the transcriptome. So I redid the alignment and it
worked, if I'm remembering correctly.
